src/components/Reown/OracleDataDisplay.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useMemo } from 'react';
+import { t } from 'ttag';
+import hathorLib from '@hathor/wallet-lib';
+import { getGlobalWallet } from '../../modules/wallet';
+
+/**
+ * Component for displaying oracle data in a structured format
+ */
+export const OracleDataDisplay = ({ oracleData }) => {
+  // Try to parse oracle as an address
+  const parsedOracleInfo = useMemo(() => {
+    try {
+      const wallet = getGlobalWallet();
+      if (!wallet || !oracleData.oracle) {
+        return { raw: oracleData.oracle, isAddress: false };
+      }
+
+      // Convert hex string to Buffer
+      const oracleBuffer = Buffer.from(oracleData.oracle, 'hex');
+      const network = wallet.getNetworkObject();
+
+      // Try to parse using available script parsing methods
+      if (hathorLib.scriptsUtils && hathorLib.scriptsUtils.parseScript) {
```

Comment:
Why do you need this `&&`?

Reply:
suggestion(minor): Just a syntatic sugar to achieve the same results
```suggestion
      if (hathorLib.scriptsUtils?.parseScript) {
```